### PR TITLE
Provide optional access to existing bindings on Injector

### PR DESCRIPTION
**Pull request by jw3**
Please consider this change which adds optional access to existing bindings on an Injector.

### CODE REVIEW

File: src/test/scala/net/codingwell/scalaguice/InjectorExtensionsSpec.scala [hunk 1]
```diff
@@ -53,5 +53,14 @@ class InjectorExtensionsSpec extends WordSpec with Matchers {
     "allow instance to be retreived using a type parameter and an annotation class" in {
       injector.instance[B, Named]
     }
+
+    "allow existing bindings to be retrieved optionally" in {
+      val Some(binding) = injector.existingBinding[A]
+      binding.getKey.getTypeLiteral == typeLiteral[B]
```
Comment (jw3) [inline src/test/scala/net/codingwell/scalaguice/InjectorExtensionsSpec.scala]: Well that doesn't make much sense :/ 

Ill fix that up...

File: src/main/scala/net/codingwell/scalaguice/InjectorExtensions.scala [hunk 2]
```diff
@@ -25,5 +25,7 @@ object InjectorExtensions {
     def instance[T: Manifest] = i.getInstance(typeLiteral[T].toKey)
     def instance[T: Manifest](ann: Annotation) = i.getInstance(typeLiteral[T].annotatedWith(ann))
     def instance[T: Manifest, Ann <: Annotation : Manifest] = i.getInstance(typeLiteral[T].annotatedWith[Ann])
+
+    def existingBinding[T: Manifest] = Option(i.getExistingBinding(typeLiteral[T].toKey))
```
Comment (nbauernfeind) [inline src/main/scala/net/codingwell/scalaguice/InjectorExtensions.scala]: Please add the annotation variants as well. (i.e. the other variants of the 'instance' methods)

**Comment by nbauernfeind** [comment]
Aside from the comment, this is looking great.

**Comment by tsuckow** [comment]
Thanks for spear heading this Nate. It will probably be next week before I can review since I'll be busy with barns& noble maker faire this weekend.

File: src/main/scala/net/codingwell/scalaguice/InjectorExtensions.scala (hunk 2 above)
Comment (jw3) [inline src/main/scala/net/codingwell/scalaguice/InjectorExtensions.scala]: Good call.

**Comment by jw3** [comment]
Updated, added annotation variants.

**Comment by nbauernfeind** [comment]
This looks great to me. I don't have merge/release credentials, so you'll have to wait for Thomas to get it into maven central.

**Comment by jw3** [comment]
Ok, great.  Thanks

**Comment by tsuckow** [comment]
Published 4.0.1-SNAPSHOT, take it for a spin and I'll see about a release next week if you give it the thumbs up.

**Comment by jw3** [comment]
Looks good other than I found an additional use case requiring the key to be computed externally, ie
```scala
def existingBinding[T: Manifest](key: => Key[T])
```

Submit new PR, or reopen this one?

**Comment by tsuckow** [comment]
Either way should be fine as long as you don't rewrite history.

**Comment by jw3** [comment]
#52 submitted, thanks.

**Pull request merged**